Issue 4053 - `svg-icon-maxi` toolbar upload button, hide upload popover when it overlaps content in `svg, image, video-maxi`

Fybex opened the pull request:
# Description

<!---
Please include a summary of the changes and the related issue.
Please also include relevant motivation and context.
--->


Fixes #4053 

# How Has This Been Tested?
Checked if it's possible to upload icon from toolbar upload button. 
Checked if upload popover hides when it overlaps content in these cases:
- on width fit-content enable
- when canvas width is too close to icon width
- when parent block width it too close to icon width

P.S. by upload popover I mean this:
<img width="144" alt="image" src="https://user-images.githubusercontent.com/46112160/209677176-d1c1b519-56fe-4a1b-9a12-36c4e94305a7.png">

P.S.2 Here my small [playground code editor](https://github.com/maxi-blocks/maxi-blocks/files/10308547/svg-upload-button-playground.txt), which, maybe, somehow will help you to test some cases

<!---
Please describe the tests that you ran to verify your changes.
Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration.
--->

# Test checklist

<!--- Please remove the unnecessary checkbox --->

**_ Front/Back Testing _**

-   [ ] Test the toolbar upload button
-   [ ] Test if upload popover hides in case it overlaps content (responsive, on highest hierarchy level, inside parent, etc)

**_ Pre-Code Testing _**

-   [ ] Test the toolbar upload button
-   [ ] Test if upload popover hides in case it overlaps content (responsive, on highest hierarchy level, inside parent, etc)
-   [ ] Check no commented code and no unnecessary imports
-   [ ] Standards of the project have been followed
-   [ ] No errors/warnings on console

# Checklist

-   [X] My code follows the style guidelines of this project
-   [X] I have performed a self-review of my code
-   [X] My changes generate no new warnings/errors

Fybex committed:
Add upload button to `svg-icon-maxi`

Fybex committed:
Hide instead of move upload popover

Fybex committed:
Improve `isSmall`  calculation algorithm

Marko64 reviewed (changes requested):
https://user-images.githubusercontent.com/19425503/209684848-18dc9f34-0acf-4c96-9942-3454657b3a1f.mp4

Fybex committed:
Create `getIsSmall` helper, change calculation algo

Fybex committed:
Merge branch 'master' into issue-4053

Fybex committed:
`MaxiPopoverButton`: replace `select` with `useSelect`

Fybex committed:
`MaxiPopoverButton` `shouldDisplayComponent`

apply hide when overlap feature for `svg-icon-maxi`, `image-maxi` and `video-maxi`

Olekrut reviewed (changes requested):
check for video-block after creating, upload button is not working
 https://jam.dev/c/a68279c0-930e-4499-a176-bb4e403e6527

Fybex committed:
Show `MaxiPopoverButton` when no content

Fybex committed:
Replace `resizerWidth` and `isImage` with `prefix`

Marko64 reviewed (changes requested):
For all three blocks, we also want the icon to disappear only if the block size is small. When the block is big, it's ok if the icon overlaps it because you can see the block clearly:

![Screenshot_3](https://user-images.githubusercontent.com/19425503/209814882-27b80bbe-fe65-49f1-b906-8a4ed4d171da.jpg)

We want to prevent this tho, when the block is small so that the icon doesn't cover most of it:
![Screenshot_4](https://user-images.githubusercontent.com/19425503/209814954-b5486ca5-4bc8-451b-af9c-0a02a362b0f7.jpg)

Fybex committed:
Hide icon, not remove component to make `children` work

Fybex committed:
Fix confusion with naming

Fybex committed:
Merge branch 'master' into issue-4053

Fybex committed:
Remove useless `useEffect`
